File: 326.PowerOf3/Eric.cpp

// Name: Eric Kwon
// Date: October 4th, 2017
// From: LeetCode
// Problem: #326 Power of Three

class Solution {
public:
    bool isPowerOfThree(int n) {
        
        // Find the maximum integer power of 3 value
        // Which is 3^19
        int maxPower3 = pow(3,19);
        
        // If 0 or less return false
        if (n <= 0)
            return false;

        // If the maxPower3 % input is 0, it's also power of 3
        else if (maxPower3 % n == 0)
            return true;

        // Otherwise return false;
        else
            return false;
    }
};

File: 217.ContainsDupe/Eric.java

// Name: Eric Kwon
// Date: October 18th, 2017
// From: LeetCode
// Problem: #217 Contains Duplicate

class Solution {
    public boolean containsDuplicate(int[] nums) {
        
        // Use HashSet which will add all the array elements
        Set<Integer> list = new HashSet<Integer>();
        int arraySize = nums.length;
        
        // Traverse through the array
        for (int i=0; i < arraySize ; i++) {
            
            // If same element has been added to hashset, return true
            if (list.contains(nums[i]))
                return true;
            
            // Otherwise add onto hashset
            list.add(nums[i]);
        }
        
        // If none, return false
        return false;
    }
}